csrc/instance_norm_nvfuser_kernel.cu

```diff
@@ -0,0 +1,269 @@
+#include <iostream>
+#include <map>
+#include <vector>
+
+#include <torch/extension.h>
+
+#include <torch/csrc/jit/codegen/cuda/kernel_cache.h>
+#include <torch/csrc/jit/codegen/cuda/ops/all_ops.h>
+
+// Hashing machinery for Params
+// Fowler–Noll–Vo hash function
+// see https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function
+template <typename Params>
+struct ParamsHash {
+  // Params must be a POD because we read out its memory
+  // contenst as char* when hashing
+  static_assert(std::is_pod<Params>::value, "Params is not POD");
+
+  size_t operator()(const Params& params) const {
+    auto ptr = reinterpret_cast<const uint8_t*>(&params);
+    uint32_t value = 0x811C9DC5;
+    for (int i = 0; i < (int)sizeof(Params); ++i) {
+      value ^= ptr[i];
+      value *= 0x01000193;
+    }
+    return (size_t)value;
+  }
+};
+
+template <typename Params>
+struct ParamsEqual {
+  // Params must be a POD because we read out its memory
+  // contenst as char* when comparing
+  static_assert(std::is_pod<Params>::value, "Params is not POD");
+
+  bool operator()(const Params& a, const Params& b) const {
+    auto ptr1 = reinterpret_cast<const uint8_t*>(&a);
+    auto ptr2 = reinterpret_cast<const uint8_t*>(&b);
+    return memcmp(ptr1, ptr2, sizeof(Params)) == 0;
+  }
+};
+
+using namespace torch::jit::fuser::cuda;
+using namespace at::indexing;
+
+// Make a tensor that is known to be fully contiguous of dimensionality=ndims,
+// but unknown sizes
+TensorView* makeContigTensor(size_t ndims, DataType dtype = DataType::Float) {
```

Comment:
[memo for me] comes from https://github.com/pytorch/pytorch/blob/127c9402d0ae7e3f72aa8ec18dbb9b8f12089bae/benchmarks/cpp/nvfuser/utils.h#L37